File: internal/indexer/extract/generic/testdata/php/sample.php
<?php

namespace App\Store;

use App\Base\BaseRepo;
use App\Contracts\{Repo, Cache as LruCache};
use function App\Util\helper;

/**
 * A user repository.
 */
#[Service]
final class UserRepo extends BaseRepo implements Repo, \Countable
{
    use Logging;

    public const LIMIT = 10;
    private array $users = [];
    protected static int $count = 0;

    public function __construct(private Db $db)
    {
        parent::__construct();
    }

    public function find(string $id): ?User
    {
        Helper::check($id);
        $u = new User($id);
        $this->log("find");
        return helper($id);
    }

    /** @deprecated */
    abstract protected function legacy(): void;
}

interface Repo
{
    public function find(string $id): ?User;
}

trait Logging
{
    public function log(string $m): void {}
}

enum Mode: string
{
    case Read = 'r';
    case Write = 'w';
}

function top_level(int $x): int
{
    return helper($x);
}

File: internal/indexer/extract/generic/testdata/java/sample.java
package com.example.store;

import java.util.List;
import java.util.*;
import static java.lang.Math.max;
import com.example.base.BaseRepo;

/**
 * Repository for users.
 */
@Service
public class UserRepo extends BaseRepo<User> implements Repo, AutoCloseable {
    private static final int LIMIT = 10;
    protected List<User> users;
    String name;

    public UserRepo(List<User> users) {
        super();
        this.users = users;
    }

    @Override
    public User find(String id) {
        Helper.check(id);
        User u = new User(id);
        return users.get(max(0, LIMIT));
    }

    @Deprecated
    static void legacy() {}

    private abstract static class Inner implements Runnable {}

    public enum Mode { READ, WRITE }

    interface Callback {
        void done(int code);
    }

    public record Pair(String a, int b) {}
}
